### ISSUE TITLE
When I use Flutter to retrieve the volume of an iOS device, the sound being played on the device switches from the speaker to the earpiece, and this only happens when the app is launched. What could be the reason for this?

### ISSUE THREAD

**Issue by Wing-Li** [yosemiteyss/flutter_volume_controller#53]
When I use Flutter to retrieve the volume of an iOS device, the sound being played on the device switches from the speaker to the earpiece, and this only happens when the app is launched. What could be the reason for this?

**Comment by yosemiteyss** [comment]
Could you provide some code sample that can produce this issue? Which plugin version and iOS version are you using?

**Comment by Wing-Li** [comment]
## Environment:

#### Flutter
~~~
[✓] Flutter (Channel stable, 3.10.1, on macOS 13.2.1 22D68 darwin-arm64, locale zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2022.2.3)
[✓] VS Code (version 1.78.2)
[✓] Connected device (3 available)
~~~

#### iOS:
Minimum system version: 13.0
Deployment Target: 13.0
platform :ios, '13.0'

#### Devices:
iPhone 12 - 16.1
iPhone 12 Pro Max - 16.5.1

***

### My Code:

After a few seconds of app startup, I simply retrieve the current volume level. However, the music played through the speaker is switched to the earpiece. The code is as follows:

~~~
// Causing the speaker to change
static Future<String> getVolumeLevel() async {
  double? volume = await FlutterVolumeController.getVolume();
  return volume == null ? "Failed to retrieve" : "${(volume * 100).toInt()}%";
}
~~~


And I'm sure it's this piece of code causing the issue!!! Because if I change it to the following code, the speaker works fine.

~~~
// Working properly
static Future<String> getVolumeLevel() async {
  // double? volume = await FlutterVolumeController.getVolume();
  // return volume == null ? "Failed to retrieve" : "${(volume * 100).toInt()}%";
  return "Failed to retrieve";
}
~~~

***

I'm really frustrated~~~

**Comment by yosemiteyss** [comment]
Fixed in version 1.2.7.

**Comment by yosemiteyss** [comment]
Fixed in version 1.2.7.

**Comment by Wing-Li** [comment]
> Fixed in version 1.2.7.

Yay, it's working smoothly now! You're really amazing.
